src/lib/components/ui/ArticleList.svelte

```diff
@@ -82,23 +94,38 @@
 		class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-y-4 md:gap-y-10 gap-x-6 justify-center"
 	>
 		{#each filteredArticles.slice(0, visibleArticles) as article, index}
-			{#if index === visibleArticles - ARTICLES_PER_PAGE}
-				<div id={`article-${index}`} bind:this={newArticleRef}>
-					<ArticleCard {article} />
-				</div>
-			{:else}
-				<div id={`article-${index}`}>
-					<ArticleCard {article} />
-				</div>
-			{/if}
+			<div transition:slide={{ delay: 100 * (index % ARTICLES_PER_PAGE) }}>
+				<ArticleCard {article} />
+			</div>
 		{/each}
+
+		{#if loading}
+			{#each Array(ARTICLES_PER_PAGE) as _, i (i)}
+				<div class="flex flex-col justify-center h-fit animate-pulse">
+					<div class="flex flex-col w-full">
+						<div class="aspect-square w-full bg-gray-200 rounded-md"></div>
+					</div>
+					<div class="flex flex-col py-6 w-full space-y-4">
+						<div class="flex gap-1 items-start w-full text-sm">
+							<div class="w-16 h-6 bg-gray-200 rounded-md"></div>
+							<div class="w-16 h-6 bg-gray-200 rounded-md"></div>
+						</div>
+						<div class="h-8 bg-gray-200 w-3/4 rounded-md"></div>
+						<div class="h-4 bg-gray-200 w-full rounded-md"></div>
+						<div class="h-4 bg-gray-200 w-5/6 rounded-md"></div>
+						<div class="h-4 bg-gray-200 w-1/2 rounded-md"></div>
+					</div>
+				</div>
```

Comment:
you should create a local [snippet](https://svelte.dev/docs/svelte/snippet), name it `cardSkeleton` and use like this `{@render cardSnippet()}`